tests/unit/serialization/serialization_variant.cpp

```diff
@@ -0,0 +1,87 @@
+//  Copyright (c) 2015 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/runtime/serialization/serialize.hpp>
+#include <hpx/runtime/serialization/string.hpp>
+#include <hpx/runtime/serialization/variant.hpp>
+
+#include <hpx/runtime/serialization/input_archive.hpp>
+#include <hpx/runtime/serialization/output_archive.hpp>
+
+#include <hpx/util/lightweight_test.hpp>
+
+template <typename T>
+struct A
+{
+    A() {}
+
+    A(T t) : t_(t) {}
+    T t_;
+
+    A & operator=(T t) { t_ = t; return *this; }
+
+    friend bool operator==(A a, A b)
+    {
+        return a.t_ == b.t_;
+    }
+
+    friend std::ostream& operator<<(std::ostream& os, A a)
+    {
+        os << a;
+        return os;
+    }
+
+    template <typename Archive>
+    void serialize(Archive & ar, unsigned)
+    {
+        ar & t_;
+    }
+};
+
+int main()
+{
+    std::vector<char> buf;
+    hpx::serialization::output_archive oar(buf);
+    hpx::serialization::input_archive iar(buf);
+
+    boost::variant<int, std::string, double, A<int> > ovar = std::string{"dfsdf"};
```

Comment:
The use of list-initialization here is going to cause our intel13 builders to complain.